src/packages/mikroorm/src/introspection/generate.ts

```diff
@@ -133,6 +133,31 @@ const convertToCamelCasePropertyNames = (metadata: EntityMetadata[]): void => {
 	}
 };
 
+const assertUniqueForeignKeys = (table: DatabaseTable): void => {
+	const uniqueForeignKeys = new Set();
+	const definedForeignKeys = Object.values(table.getForeignKeys());
+
+	for (const foreignKey of definedForeignKeys) {
+		const { localTableName, referencedTableName, columnNames, referencedColumnNames } = foreignKey;
+		const serializedValue = JSON.stringify({
```

Comment:
`JSON.stringify` is deterministic as long as the properties were created in the same order on the object, which in this case they will be, so all good, just noting after doing a bit of digging.

Reply:
yeah I had the same thought....

Is there a easy way or do we literally need to sort it first for each array?

Reply:
they are just string arrays so we should be ok with adding just sort.